hazelcast/src/main/java/com/hazelcast/map/impl/query/QueryOperation.java

```diff
@@ -123,28 +78,8 @@ public ExceptionAction onInvocationException(Throwable throwable) {
         return super.onInvocationException(throwable);
     }
 
-    @Override
-    public void onExecutionFailure(Throwable e) {
-        if (isNativeInMemoryFormat()) {
-            // This is required since if the returnsResponse() method returns false there won't be any response sent
-            // to the invoking party - this means that the operation won't be retried if the exception is instanceof
-            // HazelcastRetryableException
-            sendResponse(e);
-        } else {
-            super.onExecutionFailure(e);
-        }
-    }
-
-    @Override
```

Comment:
Not needed anymore. The operation.call return status indicates there is or is not a response.